test/test_cmrdr.cpp

```diff
@@ -31,462 +31,34 @@
- * If modules are supported in all slots > 0, then this is clearly overkill.
- * In that case the application must build a database conting information about which (sub-)modules are in where.
- */
-static uint32_t            cfg_module_ident_numbers[] =
-{
-    /* Order is not important */
-    0x00000001,
-    0x00000032,
-};
+#include "utils_for_testing.h"
 
-typedef struct cfg_submodules
-{
-   uint32_t                module_ident_number;
-   uint32_t                submodule_ident_number;
-   pnet_submodule_dir_t    direction;
-   uint16_t                input_length;
-   uint16_t                output_length;
-} cfg_submodules_t;
-
-static cfg_submodules_t    cfg_submodules[4];
-
-static int my_connect_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   pnet_result_t *p_result);
-static int my_release_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   pnet_result_t *p_result);
-static int my_dcontrol_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   pnet_control_command_t control_command,
-   pnet_result_t *p_result);
-static int my_ccontrol_cnf(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   pnet_result_t *p_result);
-static int my_state_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   pnet_event_values_t state);
-static int my_read_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   uint16_t api,
-   uint16_t slot,
-   uint16_t subslot,
-   uint16_t idx,
-   uint16_t sequence_number,
-   uint8_t **pp_read_data, /* Out: A pointer to the data */
-   uint16_t *p_read_length, /* Out: Size of data */
-   pnet_result_t *p_result); /* Error status if returning != 0 */
-static int my_write_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   uint16_t api,
-   uint16_t slot,
-   uint16_t subslot,
-   uint16_t idx,
-   uint16_t sequence_number,
-   uint16_t write_length,
-   uint8_t *p_write_data,
-   pnet_result_t *p_result);
-static int my_exp_module_ind(
-   pnet_t *net,
-   void *arg,
-   uint16_t api,
-   uint16_t slot,
-   uint32_t module_ident);
-static int my_exp_submodule_ind(
-   pnet_t *net,
-   void *arg,
-   uint16_t api,
-   uint16_t slot,
-   uint16_t subslot,
-   uint32_t module_ident,
-   uint32_t submodule_ident);
-static int my_new_data_status_ind(
-   pnet_t *net,
-   void *arg,
-   uint32_t arep,
-   uint32_t crep,
-   uint8_t changes,
-   uint8_t data_status);
-
-static void test_periodic(os_timer_t *p_timer, void *p_arg)
-{
-   /* Set new output data every 10ms */
-   tick_ctr++;
-   if ((main_arep != 0) && (tick_ctr > 10))
-   {
-      tick_ctr = 0;
-      data[0] = data_ctr++;
-      pnet_input_set_data_and_iops(g_pnet, 0, 1, 1, data, sizeof(data), PNET_IOXS_GOOD);
-   }
-
-   pnet_handle_periodic(g_pnet);
-}
 
-class CmrdrTest : public ::testing::Test
+class CmrdrTest : public IntegrationTest
 {
 protected:
-   virtual void SetUp()
+
+   virtual void SetUp() override
    {
       mock_init();
-      init_cfg();
-      counter_reset();
+      cfg_init();
```

Comment:
Can we move more common code into the base class? The init functions for instance seems to be called in most/all tests?

Reply:
The idea was to be able to easily adjust the setup for different test cases. So far mock_clear() is really the only one that differ between the different cases. I assume that we would like to test with different configurations in different tests (not yet implemented).

Reply:
But isn't there a common baseline that all tests would need? Resetting of mock data for instance. A test that has other needs can override the base init if required.

Reply:
Moved more functionality to base class.